walk/config.go

```diff
@@ -73,7 +86,7 @@ func (*Configurer) RegisterFlags(fs *flag.FlagSet, cmd string, c *config.Config)
 func (*Configurer) CheckFlags(fs *flag.FlagSet, c *config.Config) error { return nil }
 
 func (*Configurer) KnownDirectives() []string {
-	return []string{"exclude", "follow", "ignore"}
+	return []string{"generation_mode", "exclude", "follow", "ignore"}
```

Comment:
This is the directive named used by the Aspect plugins today. I think we need something better here that doesn't include the word "mode" that already has a meaning within gazelle.

Ideas?

Reply:
build_file_action?
create_build_files yes/no?

Reply:
I am going to try to this patch at Airtable. I am thinking instead of manually specifying `generation_mode` to isntead do something like (pseudocode) `collectionOnly := f == nil !any(file.endswith('.go') for file in ents)`, since we have go code spread throughout the tree

Reply:
I think different repos will have different preferences and we can't really make any assumptions based on which files exist.

Reply:
Assuming you are in David's situation, could you still leverage `generation_mode: update` to simplify your language implementation or would you need to resort to `create` with the same hack mentioned in the linked issue? 

I'm asking because adding a new directive comes with a complexity cost that may not be justified if it doesn't address many use cases. Do you see a way to simplify what a language would need to do to get this behavior without adding a new directive?

Reply:
I've changed my hack described in the issue to [a patch](https://github.com/aspect-build/aspect-cli/blob/main/patches/bazelbuild_bazel-gazelle_aspect-walk-subdir.patch) that's essentially the same as this PR. The hack required manually walking the fs (in addition to gazelle walking it) which is bad in many ways, performance being a big one.

I think no matter what a new API is needed so gazelle can generate the `language.GenerateArgs` with the correct files. It could be an API in the `Configure` phase, but today there are no APIs there that change gazelle's behaviour and adding such an API sounds far more complex then adding a directive.